plugins/api/plugin.js

```diff
@@ -0,0 +1,122 @@
+/* eslint no-console: [ 'error', { allow: [ 'error' ] } ] */
+
+/**
+ * WordPress dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+
+/* External dependencies */
+import { isFunction } from 'lodash';
+
+/**
+ * Singleton class for registering plugins.
```

Comment:
Why did you choose a singleton pattern? Why the divergence from blocks registration?

Do you plan to follow-up with necessary changes to reinstate consistency if this is the path we go?

Reply:
We've also pondered on at least one occasion whether the data backing of blocks should just be `@wordpress/data`. Adding yet another approach into the mix is worsening this consistency.

(If it was data-backed, at least I'd be more comfortable with `isPluginSidebarOpened` operating on the registry, since at that point it's more like composing selectors from other stores)